openapi/sms/v1/partner-service.yaml

```diff
@@ -266,149 +229,170 @@ components:
+          enum: [missing, invalid]
+          example: invalid
+          description: 'Type of the error.'
       required:
-        - source
-        - destination
-        - message
-        - emarsysMessageId
-        - isTransactional
+        - fieldName
+        - description
+        - errorType
+    ClientConfigurationErrors:
+      type: object
+      properties:
+        errors:
+          type: array
+          items:
+            $ref: '#/components/schemas/ClientConfigurationError'
+            minimum: 1
+    # Test Connection
     TestConnectionRequest:
       type: object
       properties:
         customFieldValues:
           $ref: '#/components/schemas/CustomFieldValues'
       required:
         - customFieldValues
-    TestConnectionResponseOK:
+    TestConnectionSuccessResponse:
       type: object
       properties:
         state:
           type: string
           enum: [success]
           example: 'success'
+          description: 'Connection test was successful when state is "success".'
       required:
         - state
-    TestConnectionResponseKO:
+    TestConnectionFailureResponse:
       type: object
       properties:
         state:
           type: string
           enum: [error]
           example: 'error'
+          description: 'Connection test was unsuccessful when state is "error".'
         description:
           type: string
           example: 'There was an error'
+          description: 'Description of the error.'
         errorType:
           type: string
           enum: [temporary, credential, integration]
           example: temporary
+          description: |
+            | Value | Description |
+            |---|---|
+            | temporary | There was a temporary issue whilst performing the request so try again later |
+            | credential | There was a problem authorizing the client on the vendor platform |
+            | integration | There is an issue related to the integration between Emarsys and the Partner service |
       discriminator:
         propertyName: state
       required:
         - state
         - description
-        - isTemporary
+        - errorType
     TestConnectionErrors:
       type: object
       properties:
         errors:
           type: array
           items:
-            $ref: '#/components/schemas/ClientConfigurationError'
+            $ref: '#/components/schemas/TestConnectionError'
             minimum: 1
-    SMSTestConnectionError:
+    TestConnectionError:
       type: object
       properties:
         fieldName:
           type: string
           example: 'servicePlanId'
+          description: 'Key to uniquely identify the field.'
         description:
           type: string
           example: 'servicePlanId can only contain numbers.'
+          description: 'Description of the error.'
       required:
         - fieldName
         - description
-    SMSSendingPartnerError:
+    # Outbound Messages
+    OutboundMessage:
```

Comment:
I think SMS could be somehow included in the name because we might support MMS in the future.

Reply:
hmm how about, for the future, `OutboundMultimediaMessage` would be an `OutboundMessage` with an extra `mediaUrl`or `mediaUrls` property? it is pretty much a message with an extra URL(s) for the media

Reply:
I am fine with that if you think that SMS messages will not have any properties that MMS messages do not have